src/evict/evict_lru.c

```diff
@@ -1942,6 +1964,41 @@ __evict_get_min_pages(WT_SESSION_IMPL *session, uint32_t target_pages)
     return (min_pages);
 }
 
+/*
+ * __try_restore_walk_position --
+ *     Try to restore the walk position from saved soft pos. Returns true if the walk position is
+ *     restored.
+ */
+static WT_INLINE int
+__try_restore_walk_position(WT_SESSION_IMPL *session, WT_BTREE *btree, uint32_t walk_flags)
+{
+#define PATH_STR_MAX 1024
+    char path_str[PATH_STR_MAX];
+    size_t path_str_offset;
+    double unused; /* GCC fails to WT_UNUSED() :( */
+
+    if (btree->evict_ref != NULL)
+        return (0); /* We've got a pointer already */
+    if (btree->evict_pos < 0.)
+        return (0); /* No restore point */
+    WT_RET_ONLY(
+      __wt_page_from_npos_for_eviction(session, &btree->evict_ref, 0, walk_flags, btree->evict_pos),
```

Comment:
TODO: Record the saved ref as an opaque ID and compare it with the restored one. Add a counter for whenever these don't match.